python/ray/serve/tests/test_deployment_in_container.py

```diff
@@ -0,0 +1,256 @@
+import asyncio
+import os
+import re
+import sys
+from pathlib import Path
+
+import numpy as np
+import pytest
+
+import ray
+from ray import serve
+from ray._private.state_api_test_utils import verify_failed_task
+from ray._private.test_utils import wait_for_condition
+from ray.serve.handle import DeploymentHandle
+from ray.tests.conftest import *  # noqa
+from ray.util.scheduling_strategies import PlacementGroupSchedulingStrategy
+from ray.util.state import list_tasks, list_workers
+
+# Runtime env that points to an image that layers on top of the
+# rayproject/ray:nightly-py38-cpu image and downgrades python version
+# to 3.8.16 to match that of the Ray CI environment
+PLAIN_CONTAINER_RUNTIME_ENV = {
+    "container": {
+        "image": "zcin/ray:nightly-py3816-cpu",
```

Comment:
Are these public Docker images (we need to make sure there are no credentials leaked in either of them)?

Reply:
These simply use `rayproject/ray:nightly-py38-cpu` as the base image, downgrade python version to 3.8.16, then write "hello world" to a file for testing.